torch_xla/csrc/runtime/ifrt_computation_client.cc

```diff
@@ -0,0 +1,643 @@
+#include "torch_xla/csrc/runtime/ifrt_computation_client.h"
+
+#include <algorithm>
+#include <unordered_set>
+#include <vector>
+
+#include "absl/strings/ascii.h"
+#include "absl/synchronization/blocking_counter.h"
+#include "absl/types/span.h"
+#include "torch_xla/csrc/runtime/computation_client.h"
+#include "torch_xla/csrc/runtime/debug_macros.h"
+#include "torch_xla/csrc/runtime/env_vars.h"
+#include "torch_xla/csrc/runtime/initialize_pjrt.h"
+#include "torch_xla/csrc/runtime/stablehlo_helper.h"
+#include "torch_xla/csrc/runtime/tf_logging.h"
+#include "torch_xla/csrc/runtime/xla_coordinator.h"
+#include "tsl/profiler/lib/traceme.h"
+#include "xla/client/xla_builder.h"
+#include "xla/client/xla_computation.h"
+#include "xla/layout_util.h"
+#include "xla/literal.h"
+#include "xla/pjrt/distributed/distributed.h"
+#include "xla/pjrt/pjrt_api.h"
+#include "xla/pjrt/pjrt_client.h"
+#include "xla/pjrt/pjrt_executable.h"
+#include "xla/python/ifrt/compiler.h"
+#include "xla/python/ifrt/memory.h"
+#include "xla/python/ifrt/sharding.h"
+#include "xla/python/pjrt_ifrt/pjrt_array.h"
+#include "xla/python/pjrt_ifrt/pjrt_client.h"
+#include "xla/python/pjrt_ifrt/xla_sharding.h"
+#include "xla/shape.h"
+
+using xla::internal::XlaBuilderFriend;
+
+namespace torch_xla {
+namespace runtime {
+
+namespace {
+
+static const std::string spmd_device_str = "SPMD:0";
+
+// Initializes a distributed runtime client if dist_service_addr is specified
```

Comment:
is this for GPU?

Reply:
@will-cromar This can be removed - the logic is encapsulated in XlaCoordinator now.

Reply:
Oops. This one should have been removed during rebasing.